### PR TITLE
net: l2: ieee802154: document non-standard KConfig options

### PR DESCRIPTION
**Pull request by ghost**
This PR documents that a few KConfig options which stand in the way of extending the IEEE 802.15.4 stack and drivers as required by #50336 will be deprecated in the foreseeable future and are no longer to be relied on by new/changed code.

Also see the IEEE802154_2015 option, that has been deprecated and removed from code in #58490.

As for the question that was asked in the initial version of this PR's description ("How can we avoid breaking proper encapsulation of PHY as opposed to MAC in the future?"): It has been agreed with the net subsystem maintainer that an RFC will be proposed to discuss architectural patterns which help us to maintain clear separation of MAC and PHY concerns while also acknowledging economical time-to-market pressures.

### CODE REVIEW

**Review by tbursztyka — DISMISSED** [review]

File: drivers/ieee802154/Kconfig
Comment (fgrandel) [reply drivers/ieee802154/Kconfig:89]: Good point. I thought a bit about it and would like to suggest that I introduce this once I provide an alternative that users can use. For now, if you want to support auto-ack or Thread 1.2 you're actually obliged to use those options.

Is there a way to signal "deprecated for development" as opposed to "deprecated for usage"?

Comment (rlubos) [reply drivers/ieee802154/Kconfig:89]: No there isn't. So perhaps we should rather deprecate existing options only when we have an alternative to use?

Comment (fgrandel) [reply drivers/ieee802154/Kconfig:89]: I agree - does it make sense to leave the TODO in there (which should be enough to alert developers) and remove the DEPRECATED word?

Comment (rlubos) [reply drivers/ieee802154/Kconfig:89]: It would make sense to me - we clearly signal the options will be removed in the future, but don't deprecate until we have an alternative to propose (or simply no longer use the option on our own).

Comment (fgrandel) [reply drivers/ieee802154/Kconfig]: I updated the PR accordingly and removed the commit deprecating IEEE802154_2015 in favor of your upcoming PR that will completely remove the option.

**Review by rlubos — APPROVED** [review]

**Comment by fgrandel** [comment]
@tbursztyka @jukkar Could you have another look? Was previously approved by @tbursztyka and just needs a second approval after an update to be merged. :-)

**Comment by jukkar** [comment]
The commit msg looks confusing as the changes do not deprecate anything but just document things to do, could that be fixed?

**Comment by fgrandel** [comment]
> The commit msg looks confusing as the changes do not deprecate anything but just document things to do, could that be fixed?

@jukkar Fixed

**Review by nashif — APPROVED** [review]

**Pull request merged**
